homeassistant/components/vconnex/__init__.py

```diff
@@ -0,0 +1,188 @@
+    DispatcherSignal,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class HomeAssistantVconnexData(NamedTuple):
+    """Home Assistant data for Vconnex domain."""
+
+    config_data: dict[str, Any]
+    device_manager: VconnexDeviceManager
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Async setup hass config entry."""
+    if DOMAIN not in hass.data:
+        hass.data.setdefault(DOMAIN, {})
+
+    vconnex_data = await init_sdk(hass, entry)
+    if vconnex_data is None:
+        return False
+
+    hass.data[DOMAIN][entry.entry_id] = vconnex_data
+
+    device_manager = vconnex_data.device_manager
+    device_manager.add_device_listener(DeviceListener(hass, device_manager))
+    update_device_registry(hass, entry, device_manager)
+
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    if unload_ok:
+        vconnex_data = hass.data[DOMAIN][entry.entry_id]
+        try:
+            vconnex_data.device_manager.release()
+        except Exception:  # pylint: disable=broad-except
+            pass
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+async def init_sdk(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> HomeAssistantVconnexData | None:
+    """Init vconnex sdk."""
+    api = VconnexAPI(
+        endpoint=DEFAULT_ENDPOINT,
+        client_id=entry.data[CONF_CLIENT_ID],
+        client_secret=entry.data[CONF_CLIENT_SECRET],
+        project_code=PROJECT_CODE,
+    )
+
+    if not await hass.async_add_executor_job(api.is_valid):
+        logger.error("Cannot connect!")
+        return None
+
+    device_manager = VconnexDeviceManager(api)
+    await hass.async_add_executor_job(device_manager.initialize)
+
+    if not device_manager.is_initialized():
+        logger.error(
+            "Could not initialize! Please reload integration or restart the server."
+        )
+        return None
+
+    hass.async_add_executor_job(
+        retrieve_device_data,
+        device_manager.device_map.values(),
+        device_manager,
+    )
+
+    config_data = dict(entry.data)
+    config_data.pop(CONF_CLIENT_SECRET, None)
+
+    return HomeAssistantVconnexData(
+        config_data=config_data, device_manager=device_manager
+    )
+
+
+def retrieve_device_data(
+    device_list: list[VconnexDevice], device_manager: VconnexDeviceManager
+):
+    """Retrieve all device data."""
+    if list is not None:
+        for device in device_list:
+            try:
+                device_manager.send_commands(
+                    device.deviceId, CommandName.GET_DATA, {"all": 1}
+                )
+            except Exception:  # pylint: disable=broad-except
+                logger.exception("Request command failure.")
+
+
+def update_device_registry(
```

Comment:
This doesn't do anything. If it's for a follow-up PR it should be removed competely until then.